### PR TITLE
Process by branch

### PR DESCRIPTION
**Pull request by marcheing**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -113,6 +113,9 @@ group :development, :test do
   gem 'i18n_generators'
 
   gem 'sprockets', '~>2.12.3' # spckets 3.0.3 breaks konacha
+
+  # Mocks and stubs for javascript tests
+  gem 'sinon-rails'
```
Comment (rafamanzo) [inline Gemfile]: Unfortunately, this gem version is outdated and no longer maintained.

Current Sinon is 1.15.0 but the gem provides 1.10.3.

What do you think about putting sinon.js on the `vendor` directory?

File: app/views/repositories/_form.html.erb [hunk 2]
```diff
@@ -94,3 +108,11 @@
   <%= f.submit t('save'), class: 'btn btn-primary' %>
   <%= link_to t('back'),  project_path(@project_id), class: 'btn btn-default' %>
 </div>
+
+<script type="text/javascript">
+  var _repository_branch = new Repository.Branch;
```
Comment (rafamanzo) [inline app/views/repositories/_form.html.erb]: I know why the var name starts with underscore. But does someone else know the reason?

File: spec/javascripts/repository/branch_spec.js.coffee [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#= require application
```
Comment (rafamanzo) [inline spec/javascripts/repository/branch_spec.js.coffee]: It's not a good practice to require the whole application.

File: Gemfile (hunk 1 above)
Comment (rafamanzo) [inline Gemfile]: Changed my mind. Adding the script to the vendor is still not good enough. The right fix is trying to reach the gem's maintainer and, if he is not going to update, fork it, release and maintain it.

(I intend to do that https://github.com/travisjeffery/sinon-rails/issues/9)

Lets keep the gem for now and, after the closing of this, we update. Are we ok with this?

File: app/views/repositories/_form.html.erb (hunk 2 above)
Comment (rafamanzo) [inline app/views/repositories/_form.html.erb]: @diegoamc , @marcheing , @danielkza , @pedroscocco ?

File: Gemfile (hunk 1 above)
Comment (pedroscocco) [inline Gemfile]: Well, it look like your ping worked.

Let's just hope he keeps it up.

File: app/views/repositories/_form.html.erb (hunk 2 above)
Comment (pedroscocco) [inline app/views/repositories/_form.html.erb]: Convention for private vars? Just a guess unfortunately

Comment (rafamanzo) [inline app/views/repositories/_form.html.erb]: Actually somehow JS vars can't hava the same name as HTML element IDs. So I had to add the underscore in order to preserve the good var name and avoid conflicts.

File: Gemfile (hunk 1 above)
Comment (rafamanzo) [inline Gemfile]: Updated our Gemfile.lock.

**Comment by rafamanzo** [comment]
I think we are ready to merge as soon as Travis' build passes.

Someone please review my latest modifications and if you agree merge.

**Pull request merged**